Limbo.Console.Generator/AutoCompletes.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Limbo.Console.Sharp.Generator
+{
+    /// <summary>
+    /// Supporting functions for parsring AutoCompletes when processing [ConsoleCommands] 
+    /// </summary>
+    public static class AutoCompletes
+    {
+        /// <summary>
+        /// Parses all of the [AutoComplete] attributes on the method, if any.
+        /// </summary>
+        /// <param name="methodSymbol"></param>
+        /// <returns></returns>
+        internal static IEnumerable<AutoCompleteDefinition> Parse(IMethodSymbol methodSymbol)
+        {
+            var autoCompletes = methodSymbol.GetAttributes()
+                                            .Where(attr => attr.AttributeClass?.Name == nameof(AutoCompleteAttribute))
+                                            .Select(AsAutoCompleteDefinition)
+                                            .ToArray();
+
+            ValidateAutoCompletes(methodSymbol.Name, autoCompletes);
+
+            return autoCompletes;
+        }
+
+        private static void ValidateAutoCompletes(string methodName, IEnumerable<AutoCompleteDefinition> autoCompletes)
+        {
+            var duplicateArgIndexes = autoCompletes
```

Comment:
Should we also validate that a [ConsoleCommand] is present on the same member?

Reply:
If limbo console doesn't throw errors from the code generated with it on something without [ConsoleCommand] I'm not sure I care either way. If it does throw errors then I think we should check this. What do you think?